core/modules/node/node.module

```diff
@@ -2268,6 +2272,16 @@ function node_page_default() {
  * @see node_menu()
  */
 function node_page_view(Node $node) {
+  // Get the current user.
+  global $user;
+  // If the user has chosen to hide this node from page views return 404.
+  $pageless = config_get('node.type.' . $node->type, 'settings.pageless');
+  $bypass_pageless = user_access('bypass pageless content restriction', $user);
+  if ($pageless && !$bypass_pageless) {
+    backdrop_not_found();
+    backdrop_exit();
```

Comment:
@quicksketch Just tried `MENU_NOT_FOUND` and no luck. This is what I get:
![screenshot from 2018-08-30 21-25-58](https://user-images.githubusercontent.com/1071483/44888336-8faba080-ac9e-11e8-8b76-9ed633c461b6.png)

Just displays the actual text "404" with the node title. I guess this is a context where it doesn't work?

Reply:
I think there is another bug in returning `MENU_NOT_FOUND`, it seems to do this anywhere it is returned. Returning `MENU_ACCESS_DENIED` works as it should though. Strange. Although I'd prefer we do it properly by returning `MENU_NOT_FOUND`, I don't really want to block this issue.

Reply:
Hm, actually I got the same thing as you until I cleared all caches, then returning `MENU_NOT_FOUND` worked as expected. Note that returning `MENU_NOT_FOUND` has to be done before `backdrop_set_title()`, otherwise the page title gets set on the 404 page.

Reply:
Not sure what I'm doing differently but still not working for me to use MENU_NOT_FOUND here.